### DISCUSSION TITLE
Allow push to wiki for other users

### DISCUSSION THREAD

**Discussion by kiozen** [community/community#22683] [New to GitHub]
<p>Hi</p>
<p>I am fairly new to Github migrating my open source project from Bitbucket to Github. Right now I struggle with the access rights to the Wiki. In the settings I allow public editing of the Wiki. This seems to work if other users edit the Wiki via the Web GUI. However if they use git to clone the Wiki they can’t push their changes. (error 403, permission denied)</p>
<p>What do I have to do to allow others to push to the Wiki?</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/kiozen">@kiozen</a>!</p>
<p>Just for clarification, are the people who are having trouble pushing to the wiki collaborators or members of the general public? It sounds like they are members of the general public, but I want to be sure.</p>
<p>Also, are you experiencing the same issues pushing to your wiki? Is there anyone else who have had success pushing to your wiki?</p>
<p>Thanks!</p>

**Comment by kiozen** [comment]
<p>Hi,</p>
<p>it’s working for collaborators but not for the general public. And I would like to allow it to the general public.</p>
<p>I can push to my project’s wiki.</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/kiozen">@kiozen</a>,</p>
<p>Thanks for the confirmation and clarification on that. When members of the general public are trying to push to the wiki, do you know if they’re authenticating to GitHub at all or are they trying to push anonymously?</p>
<p>Thanks!</p>

**Comment by kiozen** [comment]
<p>On a push git asks for login and password. If these are my credentials everything is ok. If they are from anyone else outside the project it fails.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>This is the expected behavior. A wiki can be edited via the GitHub UI by anyone with a valid GitHub login, if the repository is configured that way. But pushing to the underlying wiki repository is restricted to the people that have write access to the base repository since <code>git push</code> enables some much more destructive actions like completely rewriting the history of the repository.</p>
<p>Let us know if you have more questions.</p>

**Comment by kiozen** [comment]
<p>Thanks for clarification!</p>

**Comment by bb010g** [comment]
<aside class="quote group-github-employees quote-modified" data-username="lee-dohm" data-post="6" data-topic="2453">
<div class="title">
<div class="quote-controls"></div>
 lee-dohm:</div>
<blockquote>
<p><em>…</em><code>git push</code> enables some much more destructive actions like completely rewriting the history of the repository.</p>
</blockquote>
</aside>
<p>How would a protected–branch-like situation work, where general access could be restricted to not allow history modification?</p>

**Reply by SuibianP** [reply]
@lee-dohm I would also propose using protected branches for non-admin user (and maybe a configuration option to extend this to admins as well to prevent accidental damage).

Currently there is no way for non-admin users to programmatically interact with wiki contents.
